tests/test_diffraction_StaticStructureFactor.py

```diff
@@ -220,6 +220,30 @@ def test_repr(self):
         )
         assert str(sf) == str(eval(repr(sf)))
 
+    def test_S_0_is_N(self):
+        L = 10
+        N = 1000
+        sf = self.build_structure_factor_object(bins=100, k_max=0.1)
+        box, points = freud.data.make_random_system(L, N)
+        system = freud.AABBQuery.from_system((box, points))
+        sf.compute(system)
+        assert np.isclose(sf.S_k[0], N)
```

Comment:
This PR doesn't actually implement the proposal in issue #850. What we need is that the lowest bin _center_ is at `k_min` (its left edge would be in negative k-space) and the highest bin _center_ is at `k_max`.

> For comparison, the dynasor code puts it first bin center at 0 and its last bin center at `k_max`. That means the first bin's left edge is slightly below 0 and the last bin's right edge is slightly above `k_max`. This seems appropriate enough for our Debye code, since it only uses the bin centers.
https://github.com/glotzerlab/freud/issues/850#issuecomment-944799533

This test passes for `StaticStructureFactorDebye`, but not for the reason it should. The changes in b9a36120bc13ad31230f691c79fb66c39f66f582 make the `k_max` so small that `sf.bin_centers[0] == 0.0005`. That's close enough to zero that the Debye method gives `sf.S_k[0] == 999.99896`, which in turn passes the test `assert np.isclose(999.99896, 1000)`. If you change `k_max` back to 10, you get `sf.bin_centers[0] == 0.05` and `sf.S_k[0] == 989.6409`, which fails the test: `assert np.isclose(989.6409, 1000)`. That value of `k_max = 10` was deliberately chosen, and we should keep it there. There are other sets of parameters that can work, but `bins=100, k_max = 10` is a known good set and `bins=100, k_max=0.1` causes the test to pass but for the wrong reasons.